src/excmds.ts

```diff
@@ -961,6 +976,168 @@ export function jumpprev(n = 1) {
     })
 }
 
+/**
+ * Jumps to a local mark, a global mark, or the location before the last mark jump.
+ * [a-z] are local marks, [A-Z] are global marks and '`' is the location before the last mark jump.
+ * @param key the key associated with the mark
+ */
+//#content
+export async function jumpmark(key: string) {
+    if (key.length !== 1) {
+        return;
+    }
+    if (key === "`") {
+        return jumpBeforeJumpMark()
+    }
+    if (!/[a-z]/i.exec(key)) {
+        return;
+    }
+    if (key === key.toUpperCase()) {
+        return jumpGlobalMark(key);
+    }
+    return jumpLocalMark(key);
+}
+
+/**
+ * Jumps to a local mark.
+ * @param key the key associated with the mark
+ * @hidden
+ */
+async function jumpLocalMark(key: string) {
+    const urlWithoutAnchor = window.location.href.split("#")[0]
+    const localMarks = await State.getAsync("localMarks");
+    const mark = localMarks.get(urlWithoutAnchor)?.get(key);
+    if (mark) {
+        const currentTabId = await activeTabId();
+        state.beforeJumpMark = { scrollX: window.scrollX, scrollY: window.scrollY, tabId: currentTabId };
+        window.scrollTo(mark.scrollX, mark.scrollY);
+        return fillcmdline_tmp(3000, `Jumped to mark '${key}'`);
+    }
+    return fillcmdline_tmp(3000, `Mark '${key}' is not set`);
+}
+
+/**
+ * Jumps to a global mark. If the tab with the mark no longer
+ * exists, jumps to another tab with the same url or creates it first if such tab does not exist.
+ * @param key the key associated with the mark
+ * @hidden
+ */
+async function jumpGlobalMark(key: string) {
+    const globalMarks = await State.getAsync("globalMarks");
+    const mark = globalMarks.get(key);
+    if (!mark) {
+        return fillcmdline_tmp(3000, `Mark '${key}' is not set`);
+    }
+    const currentTabId = await activeTabId();
+    state.beforeJumpMark = { scrollX: window.scrollX, scrollY: window.scrollY, tabId: currentTabId };
+    try {
+        const tab = await getTabByTabId(mark.tabId);
+        goToTab(tab.id).then(() => {
+            browserBg.tabs.executeScript({
+                code: `window.scrollTo(${mark.scrollX},${mark.scrollY})`
+            });
+            return fillcmdline_tmp(3000, `Jumped to mark '${key}'`);
```

Comment:
If the mark is in another tab, the message will appear in the tab where the jump is made from. I think adding this in a handler for webNavigation.onCompleted or tabs.onUpdated might fix this, but I wasn't able to access the event handlers from the content script via browserBg proxy  (the shim function is `return browser[api][func](...args)` but I need something like `return browser[api][event][func](...args))`. Any ideas?

Reply:
We should probably just add an "execute ex string in tab" command and use that.